src/ess_rhat.jl

```diff
@@ -94,7 +94,7 @@ function build_cache(::FFTAutocovMethod, samples::Matrix, var::Vector)
 
     # create cache for FFT
     T = complex(eltype(samples))
-    n = nextprod([2, 3], 2 * niter - 1)
+    n = nextprod([2, 3], 2 * max(1, niter) - 1)
```

Comment:
Since we skip computation of ESS in the special case `niter <= 4` anyway, wouldn't it be easier to also skip computing `esscache` and `rel_ess_max`? Possibly it's still nice/an improvement to handle the case `niter < 1` more gracefully here.

Reply:
I refactored the code to separate allocation of outputs from the allocation of caches with computation of values. When number of iterations is too low, this allows us to directly call the mutating rhat function, which requires no caches.